src/parser.rs

```diff
@@ -1916,6 +1970,46 @@ impl Parser {
         }
     }
 
+    pub fn parse_parameter(&mut self) -> Result<Expr, ParserError> {
```

Comment:
I'm not sure what's the point of unnamed parameters in practice and why the grammar in the SQL spec looks like they are possible. Is there an implementation that supports it? BigQuery doesn't seem to, judging from their docs..

Even if so, this doesn't seem to be the right way to handle those, as data types don't have to be exactly 1 token long. I think the logic ought to use backtracking - if we can't parse the name followed by the data type, try parsing the data type alone.

Reply:
Hmm, you are right. This syntax is probably from postgres: https://www.postgresql.org/docs/9.5/sql-createfunction.html

I will just re-implement it according to BigQuery' spec then.